beacon-chain/db/state.go

```diff
@@ -191,11 +191,18 @@ func (db *BeaconDB) SaveJustifiedState(beaconState *pb.BeaconState) error {
 
 // SaveFinalizedState saves the last finalized state in the db.
 func (db *BeaconDB) SaveFinalizedState(beaconState *pb.BeaconState) error {
+	// Delete blacklisted block hashes which came before the new finalized state
+	defer func() {
```

Comment:
No need to create a new goroutine here. Please remove the `defer func() { ... }()`